hr_holidays_weekend/hr_holidays.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2010-2013 OpenERP s.a. (<http://openerp.com>).
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#    Author Andy Kanner <andy.kenner@initos.com>
+#    Author Nikolina Todorova <nikolina.todorova@initos.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm
+from openerp.tools.translate import _
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from pytz import timezone, utc
+import datetime
+import math
+
+
+class hr_holidays(orm.Model):
+
+    _inherit = 'hr.holidays'
+
+    def get_public_holidays(self, cr, uid, employee_id):
+        holidays = []
+
+        employee = self.pool.get('hr.employee').browse(cr, uid, employee_id)
+
+        hr_holiday_public_obj = self.pool.get('hr.holidays.public')
+        if employee.address_id.country_id is None:
+            country_filter = [('country_id', '=', False)]
+        else:
+            country_filter = ['|', ('country_id', '=',
+                                    employee.address_id.country_id.id),
+                                   ('country_id', '=', False)]
```

Comment:
You can simplify this expression a lot. Also, what happends if the employee has no address ?

address = employee.address_id
country = address and address.country_id
country_filter = [('country_id', 'in', [country and country.id, False])]